src/Chapter10.Tests/Listing10.23.RegisteringAFinalizerWithProcessExit.Tests.cs

```diff
@@ -1,41 +1,43 @@
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using System;
-using System.Diagnostics;
 using System.IO;
-using Chapter10.Tests.PowerShellTestsUtilities;
+using AddisonWesley.Michaelis.EssentialCSharp.Shared.Tests;
+
 using System.Text.RegularExpressions;
 
 namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter10.Listing10_23.Tests
 {
     [TestClass]
-
     public class DisposeTests
     {
+        public TestContext TestContext { get; set; } = null!; // Auto-initialized by MSTest.
+
+        static string Ps1DirectoryPath { get; } =
+            Path.GetFullPath(
+                Path.Join(IntelliTect.Multitool.RepositoryPaths.GetDefaultRepoRoot(),"src", "Chapter10"));
 
-        static string Ps1Path { get; } = Path.GetFullPath("../../../../Chapter10/Listing10.23.RegisteringAFinalizerWithProcessExit.ps1", Environment.CurrentDirectory);
-        static bool PowerShellNotAvailable = PowerShellTestsUtilities.PowerShellNotInstalled();
-        static string PowershellEnvironmentVariableName { get; set; } = "powershell";
+        static string Ps1Path { get; } = 
+            Path.GetFullPath(Path.Join(Ps1DirectoryPath, "Listing10.23.RegisteringAFinalizerWithProcessExit.ps1"), Environment.CurrentDirectory);
+
+        private const string ProjectName = "ProcessExitTestProgram.g";
 
         [ClassInitialize]
         public static void ClassInitialize(TestContext testContext)
         {
-            if (PowerShellNotAvailable) { Assert.Inconclusive("Powershell not installed"); return; }
-            
-            if (PowerShellTestsUtilities.WindowsEnvironment()) PowershellEnvironmentVariableName = "powershell";
-            else PowershellEnvironmentVariableName = "pwsh";
-            
-            string testStatus = "create";
-            Process powershell = Process.Start(PowershellEnvironmentVariableName, $"-noprofile -command \"{Ps1Path} 0 null {testStatus}\"");
-            powershell.WaitForExit();
+            string testStage = "create";
+            Assert.AreEqual<int>(0, RunPowerShellScript(testStage, out string psOutput),
+                $"Script failed with $testStage='{testStage}'. psOutput:\n{psOutput}");
+            string projectFilePath =
+                Path.Join(Ps1DirectoryPath, ProjectName, $"{ProjectName}.csproj");
+            Assert.IsTrue(File.Exists(projectFilePath),
+                $"The expected project file, '{projectFilePath}', was not created.");
         }
 
         [ClassCleanup]
         public static void RemoveProcessExitProj()
         {
-            if (PowerShellNotAvailable) { return; }
-            string testStatus = "cleanup";
-            Process powershell = Process.Start(PowershellEnvironmentVariableName, $"-noprofile -command \"{Ps1Path} 0 null {testStatus}\"");
-            powershell.WaitForExit();
+            Assert.AreEqual<int>(0, RunPowerShellScript(
+                "cleanup", out string _));
```

Comment:
This assertion doesn't actually work.
By the time the cleanup method gets invoked the all the tests in the class have already reported success/failure.

![image](https://user-images.githubusercontent.com/952248/197562138-3a1670fc-eb22-4818-937a-82d928a9115f.png)

Reply:
Agreed... Also, if debugging and you stop, cleanup won't run.  Removing the assert and adding cleanup to classinitialize as well.